gto/base.py

```diff
@@ -58,6 +58,51 @@ def dict_status(self):
         return version
 
 
+def sort_versions(
+    versions,
+    sort=VersionSort.SemVer,
+    ascending=False,
+    version="name",
+    timestamp="created_at",
+):
+    def get(obj, key):
+        if isinstance(obj, dict):
+            return obj[key]
+        if isinstance(obj, BaseModel):
+            return getattr(obj, key)
+        raise NotImplementedError("Can sort either dict or BaseModel")
+
+    sort = sort if isinstance(sort, VersionSort) else VersionSort[sort]
+    if sort == VersionSort.SemVer:
+        # sorting SemVer versions in a right way
+        sorted_versions = sorted(
+            (v for v in versions if v.is_registered),
```

Comment:
@aguschin We don't store is_registered or discovered tag with us. While using the method with dict, `v.is_registered` may throw AttributeError here.

Reply:
Ok, I see [your example repo](https://github.com/iterative/example-mlem-get-started-frozen-v0.1.7/tags) doesn't include these cases, so I think you didn't implement support for `not is_registered` versions.

Reply:
@amritghimire, I've make it infer whether version is SemVer or just a hexsha. Will it work now?

Reply:
Thanks, it works now.

Reply:
Do you need something else to be implemented now? Otherwise I'll go ahead and create new release.

Reply:
That is all @aguschin . Thanks for continuous updates to the package.